java/samples/sample-code/src/main/java/com/microsoft/semantickernel/samples/syntaxexamples/Example57_KernelHooks.java

```diff
@@ -365,4 +369,51 @@ private static void chatCompletionHook(Kernel kernel) {
         }
     }
 
+    /**
+     * Show use of hooks added on a single invocations.
+     *
+     * @param kernel
+     */
+    private static void invocationHook(Kernel kernel) {
+        // Initialize prompt
+        String functionPrompt = "Write a paragraph about hats";
+
+        var writerFunction = KernelFunctionFromPrompt.builder()
+            .withTemplate(functionPrompt)
+            .withName("Writer")
+            .withDefaultExecutionSettings(PromptExecutionSettings
+                .builder()
+                .withMaxTokens(1000)
+                .withTemperature(1)
+                .withTopP(0.5)
+                .build())
+            .build();
+
+        HookService hooks = new HookService();
+        hooks.addPreChatCompletionHook(event -> {
+            ChatCompletionsOptions options = event.getOptions();
+            List<ChatRequestMessage> messages = options.getMessages();
+
+            messages = new ArrayList<>(messages);
+            messages.add(
+                new ChatRequestSystemMessage("Use upper case text when responding to the prompt."));
+
+            return new PreChatCompletionHookEvent(
```

Comment:
I think you can remove "Hook" from these event names since hooks don't themselves have events.